### ISSUE TITLE
Entities with null as `id` are merged [@next]

### ISSUE THREAD

**Issue by Toxicable** [typeorm/typeorm#1576]
**Issue type:**
[x] bug report

**Database system/driver:**
[x] `postgres`

**TypeORM version:**
[x] `@next`

**Steps to reproduce or a small repository showing the problem:**

https://github.com/Toxicable/typeorm-issue-entities-merged

**Comment by pleerock** [comment]
Thanks for the reproduction repo. I'll be very grateful if you next time create a PR with a failing test (see my commit for example). Fixed and will be released in 0.2.0-alpha.20 soon.

**Comment by gintsgints** [comment]
Somthing like this?

```
        const newpost = new Post();
        let cat1 = new Category();
        cat1.name2 = "1";
        let cat2 = new Category();
        cat2.name = "2";
        newpost.categories = [cat1, cat2];

        const post = connection.manager.create(Post, newpost);

        expect(post.categories).to.have.length(2);
```
